packages/animations/lib/src/fade_transition.dart

```diff
@@ -0,0 +1,232 @@
+///
+/// Content below the modal is dimmed with a [ModalBarrier].
+///
+/// ```dart
+/// /// Sample widget that uses [showModalWithFadeTransition].
+/// class MyHomePage extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return Scaffold(
+///       body: Center(
+///         child: RaisedButton(
+///           onPressed: () {
+///             showModalWithFadeTransition(
+///               context: context,
+///               child: FlutterLogoModal(),
+///             );
+///           },
+///           child: Icon(Icons.add),
+///         ),
+///       ),
+///     );
+///   }
+/// }
+///
+/// /// Displays a modal with the FlutterLogo on it.
+/// class FlutterLogoModal extends StatelessWidget {
+///   const _FlutterLogoModal();
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return Column(
+///       mainAxisAlignment: MainAxisAlignment.center,
+///       children: <Widget>[
+///         Center(
+///           child: ConstrainedBox(
+///             constraints: const BoxConstraints(
+///               maxHeight: 300,
+///               maxWidth: 300,
+///               minHeight: 250,
+///               minWidth: 250,
+///             ),
+///             child: const Material(
+///               child: Center(child: FlutterLogo(size: 250)),
+///             ),
+///           ),
+///         ),
+///       ],
+///     );
+///   }
+/// }
+/// ```
+///
+/// The `context` argument is used to look up the [Navigator] for the
+/// modal. It is only used when the method is called. Its corresponding widget
+/// can be safely removed from the tree before the modal is closed.
+///
+/// The `useRootNavigator` argument is used to determine whether to push the
+/// modal to the [Navigator] furthest from or nearest to the given `context`.
+/// By default, `useRootNavigator` is `true` and the modal route created by
+/// this method is pushed to the root navigator.
+///
+/// If the application has multiple [Navigator] objects, it may be necessary to
+/// call `Navigator.of(context, rootNavigator: true).pop(result)` to close the
+/// modal rather than just `Navigator.pop(context, result)`.
+///
+/// The `barrierDismissible` argument is used to determine whether this route
+/// can be dismissed by tapping the modal barrier. This argument defaults
+/// to true. If `barrierDismissible` is true, a non-null `barrierLabel` must be
+/// provided.
+///
+/// The `barrierLabel` argument is the semantic label used for a dismissible
+/// barrier. This argument defaults to "Dismiss".
+///
+/// Returns a [Future] that resolves to the value (if any) that was passed to
+/// [Navigator.pop] when the modal was closed.
+///
+/// See also:
+///
+/// * [FadeModalRoute], which is the route that is built by this function.
+Future<T> showModalWithFadeTransition<T>({
+  @required BuildContext context,
+  bool barrierDismissible = true,
+  String barrierLabel,
+  bool useRootNavigator = true,
+  Widget child,
```

Comment:
I believe in the other showDialog type methods we deprecated the child in favor of a builder. We should probably just have a builder argument here?